light_curves/scale_up.md

```diff
@@ -0,0 +1,559 @@
+    "ruan": {},
+    "papers_list": {
+        "paper_kwargs": [
+            {"paper_link": "2022ApJ...933...37W", "label": "Galex variable 22"},
+            {"paper_link": "2020ApJ...896...10B", "label": "Palomar variable 20"},
+        ]
+    },
+    "SDSS": {"num": 10, "zmin": 0.5, "zmax": 2, "randomize_z": True},
+    "ZTF_objectid": {"objectids": ["ZTF18aabtxvd", "ZTF18aahqkbt", "ZTF18abxftqm", "ZTF18acaqdaa"]},
+}
+
+archives = {
+    "Gaia": {"search_radius": 2 / 3600},
+    "HEASARC": {"catalog_error_radii": {"FERMIGTRIG": 1.0, "SAXGRBMGRB": 3.0}},
+    "IceCube": {"icecube_select_topN": 4, "max_search_radius": 2.0},
+    "WISE": {"radius": 1.5, "bandlist": ["W1", "W2"]},
+    "ZTF": {"nworkers": 6, "match_radius": 2 / 3600},
+}
+
+yaml_kwargs_dict = {
+    "get_samples": get_samples,
+    "consolidate_nearby_objects": False,
+    "archives": archives,
+}
+
+helpers.scale_up.write_kwargs_to_yaml(run_id=yaml_run_id, **yaml_kwargs_dict)
+```
+
+The path to the yaml file is printed in the output above.
+You can alter the contents of the file as you like.
+To use the file, set the kwarg `use_yaml=True`.
+
+Python example for the get-sample step:
+
+```{code-cell}
+sample_table = helpers.scale_up.run(build="sample", run_id=yaml_run_id, use_yaml=True)
+```
+
+Bash example:
+
++++
+
+```bash
+$ yaml_run_id=demo-kwargs-yaml
+$ bash code_src/helpers/scale_up.sh -r "$yaml_run_id" -d "use_yaml=true" -a "core"
+```
+
++++
+
+## Appendix: What to expect
+
++++
+
+### Challenges of large-scale runs
+
++++
+
+Scaling up to large sample sizes brings new challenges.
+Even if things go smoothly, each function call may need to use a lot more resources like CPU, RAM, and bandwidth, and may take much longer to complete.
+Also, we'll want to run some functions in parallel to save time overall, but that will mean they must compete with each other for resources.
+
+These issues are complicated by the fact that different combinations of samples and archive calls can trigger different problems.
+Inefficiencies in any part of the process -- our code, archive backends, etc. -- which may have been negligible at small scale can balloon into significant hurdles.
+
+Problems can manifest in different ways.
+For example, progress may to slow to a crawl, or it may run smoothly for several hours and then crash suddenly.
+If the job is running in the background, print statements and error messages may get lost and never be displayed for the user if they are not redirected to a file.
+
++++
+
+### Needs and wants for large-scale runs
+
++++
+
+The main goal is to speed up the run, so we want to look for opportunities to parallelize.
+We can group the light_curve_generator code into two main steps: (1) gather the target object sample; then (2) generate light curves by querying the archives and standardizing the returned data.
+All of the archive calls have to wait for the sample to be available before starting, but then they can run independently in parallel.
```

Comment:
Maybe add that gathering the sample does not take long compared to the archive calls, so the real opportunity for parallelization lies in the archive calls.